### PR TITLE
ADNI_Brain_Visual_Transformer_47306725

### PR DESCRIPTION
**Pull request by ritikarana9999**
**Pull Request: ADNI_Brain_Visual_Transformer_47306725 Readme**

Summary:
This pull request aims to enhance the README documentation for the "ADNI_Brain_Visual_Transformer_47306725" project. The improvements include providing a more structured overview of the project, clarifying the Vision Transformer architecture, detailing the dataset, and addressing the challenges faced during model training. Additionally, the pull request incorporates suggestions for improving clarity and readability.

Additional Details:

- The revised README is organized into clear sections, such as "Overview," "Vision Transformers," "Setting Parameters," "Build and Train the Model," and "Conclusion."

- Detailed explanations have been added to sections like "Architecture of Vision Transformer," providing insights into the alterations made to the original Vision Transformer and how the model functions.

- The "Training and Testing Process" section has been expanded to provide a comprehensive understanding of the experimentation process with different hyperparameters and the observed results from multiple models.

- Suggestions for potential enhancements, such as considering a larger dataset or leveraging pre-trained models, have been included in the "Conclusion" section to guide future improvements.

Name: Ritika Rana
Student ID: 47306725

### CODE REVIEW

**Comment by ritikarana9999** [comment]
i didn't get any feedback till now

**Comment by shakes76** [comment]
Hi Ritika, this PR was created quite late (last week), even though I can see commits were on time. Tutor will look at this ASAP, but there may be penalties for late PR submission.

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: 
- Excellent

Commit messages: 
- Commit messages are too simple and mostly the same commit messages, needs more details
- All commits are finished within two days
- there are three screenshot of google colab commit logs

Code:
- Uses Vision transformer
- Code is submitted in **one jupyter notebook** 
- Code comments are okay
- Good design

Functionality/Performance:
- No patient-level split
- Model is overfitting
- Test ACC is 64.65% **(higher than val ACC about 55%)**
- Applied data augmentation


General comments:
**PR is in the main branch**
- Solves the problem appropriately although performance is poor
- No patient-level split so the model is overfitting. 
- The code was submitted in one jupyter notebook which does meet the requirement. You will need to keep e.g. model, train, test, data preprocessing in separate files.
- The validation loss is not decreasing and sometimes even increasing from the provided three models. There could be bugs in the code which lead the model not trained properly.
- Another part needs to be pointed out is that your val ACC is lower than your test ACC which is unusual.
- Also, why is your test set so small, we have provide a test set with 4500 images and your confusion matrix only show you used about 150 images.

**Comment by shakes76** [comment]
# Marking
Commits on time but late PR -4
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation, only as a notebook -2
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem, but has training/validation loss irregularities -1
Driver Script not present -1
File structure not present -2
Shows Usage & Demo & Visualisation & Data usage, didn't use all test images -1
Module present
Commenting minimal -1
No Data leakage, no patient level split -1
Difficulty: Hard
## Commit Log
Meaningful commit messages sometimes -1
Progressive commits used over 2 days -1
## Documentation
ReadMe acceptable/good, no usage -1
Model/technical explanation minimal -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in InCorrect Branch) -2
Feedback required, update PR to correct branch, move your README into folder and restore repo README -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
Have some feedback attempts but was unsuccessful. So no feedback marks granted.
